src/twisted/web/test/test_http2.py

```diff
@@ -2470,3 +2470,102 @@ def test_noTimeoutIfConnectionLost(self):
         # Advancing the clock should do nothing.
         reactor.advance(101)
         self.assertEqual(transport.value(), sentData)
+
+
+    def test_timeoutEventuallyForcesConnectionClosed(self):
+        """
+        When a L{H2Connection} has timed the connection out, and the transport
+        doesn't get torn down within 15 seconds, it gets forcibly closed.
+        """
+        frameFactory = FrameFactory()
```

Comment:
These 3 tests share their first ~20 lines of code.  Any chance you could put this into a function to avoid the maintenance burden of updating all of them the next time someone rolls in here?